src/_pytest/logging.py

```diff
@@ -596,10 +600,7 @@ def pytest_collection(self):
             if self.log_cli_handler:
                 self.log_cli_handler.set_when("collection")
 
-            if self.log_file_handler is not None:
-                with catching_logs(self.log_file_handler, level=self.log_file_level):
-                    yield
-            else:
+            with catching_logs(self.log_file_handler, level=self.log_file_level):
```

Comment:
I don't like that `catching_logs` implicitly logs to `LogCaptureHandler` if `self.log_file_hander ` is `None`. I would suggest to be more explicit and pass

`self.log_file_handler if self.log_file_handler else LogCaptureHandler()` as the first arg to `catching_logs`. 

Then your modification to the `catching_logs` ctx can be reverted.
WDYT?

Reply:
@thisch 
IIRC there are other places where using a new / anonymous LogCaptureHandler could be used, therefore I've changed it to handle that for the arg being None here already.  (it avoids creating it then in several places etc)
But of couse for this patch alone your suggestion might make sense / seems good probably.

Reply:
@thisch, I've updated the PR to follow your suggestion.